akka-stream/src/main/scala/akka/stream/io/SslTlsNew.scala

```diff
@@ -0,0 +1,297 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.io
+
+import java.nio.{ BufferOverflowException, ByteBuffer }
+import javax.net.ssl.SSLEngineResult.HandshakeStatus
+import javax.net.ssl.SSLEngineResult.HandshakeStatus._
+import javax.net.ssl.SSLEngineResult.Status._
+import javax.net.ssl.{ SSLEngine, SSLContext }
+
+import akka.event.LoggingAdapter
+import akka.stream.{ Attributes, Outlet, Inlet, BidiShape }
+import akka.stream.stage.{ OutHandler, InHandler, GraphStageLogic, GraphStage }
+import akka.util.ByteString
+
+import scala.annotation.tailrec
+
+object SslTlsNew {
```

Comment:
This PR is work in progress. This does not even work yet. Review it accordingly.